src/port1.0/portconfigure.tcl

```diff
@@ -527,9 +527,17 @@ proc portconfigure::get_compiler_fallback {} {
                 return {gcc-4.0}
             }
         } elseif {[vercmp $xcodeversion 3.0] >= 0} {
-            return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-clang-3.4 macports-clang-3.3}
+            if {${configure.build_arch} eq "ppc" || ${configure.build_arch} eq "ppc64"} {
+                return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-gcc-6  macports-gcc-7}
```

Comment:
I've been thinking whether this should be based configure.build_arch or os.arch. I chose build_arch, as building PPC ports on an Intel machine doesn't work reliably with clang at present.